R/classes.R

```diff
@@ -126,73 +120,74 @@ print.PartInvList <- function(x, ...) {
+    cat("\nDiscrepancy between CAI of reference vs. Efocal groups under PFI:\n")
+    print(round(x$`h CAI Ref-EF`[c(1, item_set + 1), 5:8, drop = FALSE], 3))
+    cat(paste0(dashes, "\nImpact of deleting an item on the discrepancy between CAI of\nreference vs. Efocal groups under PFI:\n"))
+    print(round(x$`delta h CAI Ref-EF (deletion)`[c(item_set), 5:8, drop = FALSE], 3))
+  }
+ # ------------------------------------------------------------------------
+  if (!x$formatted) { #
+    cat(paste0(stars, "\nAGGREGATE CLASSIFICATION ACCURACY INDICES (ACAI)\n", stars))
+    cat("\nAggregate CAI under PFI computed for item subsets:\n")
+    print(round(x$`ACAI`[c(1, item_set + 1), , drop = FALSE], 3))
+    cat("\nImpact of deleting an item on aggregate CAI under PFI:\n")
+    print(round(x$`h ACAI (deletion)`[c(item_set), , drop = FALSE], 3))
+    cat(paste0(dashes, "\nDiscrepancy between aggregate CAI under SFI vs. PFI: \n"))
+    print(round(x$`h ACAI SFI-PFI`[c(1, item_set + 1), , drop = FALSE], 3))
+    cat("\nImpact of deleting an item on the discrepancy between ACAI \nunder SFI vs. ACAI under PFI:\n")
+    print(round(x$`delta h ACAI SFI-PFI (deletion)`[c(item_set), , drop = FALSE], 3))
+    cat(paste0("\n", stars, "\nAdverse Impact (AI) ratio for item subsets by invariance condition:\n", stars, "\n"))
+    print(round(x$AI[c(1, item_set + 1), , drop = FALSE], 3))
+    cat(paste0("\n", stars, "\nCOMPARING CAI FOR REFERENCE AND (EXPECTED) FOCAL GROUPS\n", stars))
+    cat("\nDiscrepancy between CAI of reference vs. Efocal groups under PFI:\n")
+    print(round(x$`h CAI Ref-EF`[c(1, item_set + 1), , drop = FALSE], 3))
+    cat(paste0(dashes, "\nImpact of deleting an item on the discrepancy between CAI of\nreference vs. Efocal groups under PFI:\n"))
+    print(round(x$`delta h CAI Ref-EF (deletion)`[c(item_set), , drop = FALSE], 3))
+    cat(paste0("\n", stars, "\nCOMPARING CAI UNDER STRICT AND PARTIAL FACTORIAL INVARIANCE\n", stars))
+    cat("\nDiscrepancy between CAI under SFI vs. PFI for the reference group:\n")
+    print(round(x$`h CAI SFI-PFI`[[1]][c(1, item_set + 1), , drop = FALSE], 3))
+    cat("\nDiscrepancy between CAI under SFI vs. PFI for the focal group:\n")
+    print(round(x$`h CAI SFI-PFI`[[2]][c(1, item_set + 1), , drop = FALSE], 3))
+    cat(paste0(dashes, "\nImpact of deleting an item on the difference between CAI under \nSFI vs. PFI for the reference group:\n"))
+    print(round(x$`delta h SFI-PFI (deletion)`[[1]][c(item_set), , drop = FALSE], 3))
+    cat("\nImpact of deleting an item on the difference between CAI under \nSFI vs. PFI for the focal group:\n")
+    print(round(x$`delta h SFI-PFI (deletion)`[[2]][c(item_set), , drop = FALSE], 3))
+    print(x$`PartInv by groups`$reference)#$outputlist)
+    print(x$`PartInv by groups`$focal)#$outputlist[c(1, item_set+1)])
+    print(x$PartInv$strict)#$outputlist[c(1, item_set+1)])
+    print(x$PartInv$partial)#$outputlist[c(1, item_set+1)])
+  }
```

Comment:
Just want to say this is really helpful for me to understand how the class works in R :)